packages/mds-stream-processor/index.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2019 City of Los Angeles
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import logger from '@mds-core/mds-logger'
+import { asArray } from '@mds-core/mds-utils'
+import { StreamProcessorController, StreamSink, StreamSource, StreamTransform } from './@types'
+
+// StreamProcessor - Read from source, apply transform to each message, and write to sink
+export const StreamProcessor = <TMessageIn, TMessageOut>(
+  source: StreamSource<TMessageIn>,
+  transform: StreamTransform<TMessageIn, TMessageOut>,
+  sink: StreamSink<TMessageOut>
```

Comment:
Why not `sinks`?

Reply:
Is this in reference to your recent changes? I think I pulled this from a version that didn't yet have those. I will update. Good catch!